zerver/lib/users.py

```diff
@@ -1058,6 +1060,12 @@ def get_user_dicts_in_realm(
 ) -> tuple[list[RawUserDict], list[APIUserDict]]:
     if user_ids is not None:
         all_user_dicts = get_realm_user_dicts_from_ids(realm.id, user_ids)
+    elif os.environ.get("PARTIAL_USERS"):  # nocoverage
+        # This just includes bot data.
+        all_user_dicts = get_partial_realm_user_dicts(realm.id)
+        # Add current user data if not spectator
+        if user_profile is not None:
+            all_user_dicts.append(user_profile_to_user_row(user_profile))
```

Comment:
This comment and adjustment belongs in `get_partial_realm_user_dicts` from an abstractions standpoint. Moving it there and passing `user_profile` as a parameter to `get_partial_realm_user_dicts`.